### PR TITLE
[Packager] Allow user to specify a custom transformer file

### PR DESCRIPTION
**Pull request by johanneslumpe**
This is an edited re-submission of #1458 because I'm stupid.

### CODE REVIEW

File: packager/packager.js [hunk 1]
```diff
@@ -198,11 +203,15 @@ function statusPageMiddleware(req, res, next) {
 }
 
 function getAppMiddleware(options) {
+  var transformerPath = options.transformer;
+  if (!path.isAbsolute(transformerPath)) {
```
Comment (sebmck) [inline packager/packager.js]: This is only available on very recent node (0.12) and io.js versions. What's the supported version for the packager?

Comment (johanneslumpe) [inline packager/packager.js]: Oh that's odd. If we have to support older version, we might add

`path.resolve(transformerPath) === path.normalize(transformerPath)`

Comment (amasad) [inline packager/packager.js]: We want to support v0.10 (after all jest doesn't run on v0.12 yet). We have a package installed for this `absolute-path`

**Comment by amasad** [comment]
Just fix the `absolute-path` thing and this should be good to go.

**Comment by johanneslumpe** [comment]
@amasad done!

**Comment by amasad** [comment]
Great! Will do pull it. It'd be nice to add some docs to that as well :) :dancer:

**Comment by johanneslumpe** [comment]
@amasad are those docs fine?

File: docs/JavaScriptEnvironment.md [hunk 2]
```diff
@@ -17,9 +17,13 @@ When using React Native, you're going to be running your JavaScript code in two
 While both environments are very similar, you may end up hitting some inconsistencies. We're likely going to experiment with other JS engines in the future, so it's best to avoid relying on specifics of any runtime.
 
 
-## JavaScript Transforms
+## JavaScript Syntax Transformers
 
-React Native ships with many JavaScript transforms to make writing code more enjoyable. If you are curious, you can see the [implementation of all those transformations](https://github.com/facebook/jstransform/tree/master/visitors). Here's the full list:
+Syntax transformers make writing code more enjoyable by allowing you to use new JavaScript syntax without having to wait for support on all interpreters.
```
Comment (amasad) [inline docs/JavaScriptEnvironment.md]: We don't *really* have to wait on all interpreters. We can just end the sentence at `by allowing you to use new JavaScript language features`.

File: docs/JavaScriptEnvironment.md [hunk 3]
```diff
@@ -27,18 +31,26 @@ ES5
...
+* [Rest Params](https://github.com/sebmarkbage/ecmascript-rest-spread): `function(type, ...args) { }`
+* [Template Literals](http://babeljs.io/docs/learn-es2015/#template-strings): ``var who = 'world'; var str = `Hello ${who}`;``
 
 ES7
 
-* Object Spread: `var extended = { ...obj, a: 10 };`
-* Function Trailing Comma: `function f(a, b, c,) { }`
+* [Object Spread](https://github.com/sebmarkbage/ecmascript-rest-spread): `var extended = { ...obj, a: 10 };`
+* [Function Trailing Comma](https://github.com/jeffmo/es-trailing-function-commas): `function f(a, b, c,) { }`
+
```
Comment (amasad) [inline docs/JavaScriptEnvironment.md]: Awesome! Thanks for the links

File: docs/JavaScriptEnvironment.md [hunk 4]
```diff
@@ -17,9 +17,13 @@ When using React Native, you're going to be running your JavaScript code in two
 While both environments are very similar, you may end up hitting some inconsistencies. We're likely going to experiment with other JS engines in the future, so it's best to avoid relying on specifics of any runtime.
 
 
-## JavaScript Transforms
+## JavaScript Syntax Transformers
 
-React Native ships with many JavaScript transforms to make writing code more enjoyable. If you are curious, you can see the [implementation of all those transformations](https://github.com/facebook/jstransform/tree/master/visitors). Here's the full list:
+Syntax transformers make writing code more enjoyable by allowing you to use new JavaScript syntax without having to wait for support on all interpreters.
+
+As of version 0.5.0, React Native ships with the [Babel JavaScript compiler](https://babeljs.io). Check [Babel documentation](http://babeljs.io/docs/advanced/transformers/) on its supported transformations for more details.
```
Comment (amasad) [inline docs/JavaScriptEnvironment.md]: add `the`: "check the Babel documentation"

Comment (amasad) [inline docs/JavaScriptEnvironment.md]: `on its supported transformations for more details.`
to
`for its supported transformations and more details.`

File: docs/JavaScriptEnvironment.md [hunk 5]
```diff
@@ -27,18 +31,26 @@ ES5
...
 
 ES7
 
-* Object Spread: `var extended = { ...obj, a: 10 };`
-* Function Trailing Comma: `function f(a, b, c,) { }`
+* [Object Spread](https://github.com/sebmarkbage/ecmascript-rest-spread): `var extended = { ...obj, a: 10 };`
+* [Function Trailing Comma](https://github.com/jeffmo/es-trailing-function-commas): `function f(a, b, c,) { }`
+
+### Custom transformer
+You can also use a custom transformer file instead of the [default one](https://github.com/facebook/react-native/blob/master/packager/transformer.js), in case you want to modify or extend the behavior beyond babel configuration.
```
Comment (amasad) [inline docs/JavaScriptEnvironment.md]: "beyond `the` babel`"
Capitalize `babel`.

File: docs/JavaScriptEnvironment.md [hunk 6]
```diff
@@ -27,18 +31,26 @@ ES5
...
+In order to so, start the packager like this:
```
Comment (amasad) [inline docs/JavaScriptEnvironment.md]: add `do`

**Comment by amasad** [comment]
Sorry for the late reply. Just a few nits on the docs

**Comment by johanneslumpe** [comment]
@amasad updated the docs. I actually didn't write those texts, but they're changed the way you wanted now :)

**Pull request closed without merging**